File: 724. Find Pivot Index/724.js
//724. Find Pivot Index solution in JavaScript
//https://leetcode.com/problems/find-pivot-index/description/

/**
 * @param {number[]} nums
 * @return {number}
 */
 var pivotIndex = function(nums) {
    let sum = 0;
    let left_sum = 0;

    for (let i = 0; i < nums.length; i++) {
        sum += nums[i];
    }

    for (let i = 0; i < nums.length; i++) {
        if (left_sum == sum - left_sum - nums[i]) return i;

        left_sum += nums[i];
    }

    return -1;
};

File: 724. Find Pivot Index/724.go
//724. Find Pivot Index solution in Golang
//https://leetcode.com/problems/find-pivot-index/description/

package main

func pivotIndex(nums []int) int {
	sum, leftSum, length := 0, 0, len(nums)

	for n := 0; n < length; n++ {
		sum = sum + nums[n]
	}

	for n := 0; n < length; n++ {
		if leftSum == sum-leftSum-nums[n] {
			return n
		}

		leftSum = leftSum + nums[n]
	}

	return -1
}
